pkg/autodetect/main.go

```diff
@@ -147,6 +154,41 @@ func (b *Background) detectClusterRoles() {
 	}
 }
 
+func (b *Background) cleanDeployments() error {
+	log.Info("Detecting orphaned deployments.")
+	deployments := &appsv1.DeploymentList{}
+	opts := &client.ListOptions{}
+	instances := &v1.JaegerList{}
+
+	// get all deployments
+	if err := b.cl.List(context.Background(), opts, deployments); err != nil {
+		return err
+	}
+
+	// get all jaeger instances
+	if err := b.cl.List(context.Background(), opts, instances); err != nil {
+		return err
+	}
+	log.Info(len(deployments.Items))
+
+	// check deployments to see which one needs to be cleaned.
+	for _, dep := range deployments.Items {
+		if _, ok := dep.Annotations[inject.Annotation]; ok {
+			jaeger := inject.Select(&dep, instances)
+			if jaeger == nil { // Jaeger instance not exist anymore, we need to clean this up.
+				inject.CleanSidecar(&dep)
+				if err := b.cl.Update(context.Background(), &dep); err != nil {
+					log.WithFields(log.Fields{
+						"deploymentName":      dep.Name,
+						"deploymentNamespace": dep.Namespace,
+					}).Error(err)
```

Comment:
Looks like it's missing the actual log message? :) 

`.Info("failed to clean sidecar")`